app/models/container_service.rb

```diff
@@ -27,10 +27,23 @@ def container_groups_count
     number_of(:container_groups)
   end
 
+  include EventMixin
   include Metric::CiMixin
 
   PERF_ROLLUP_CHILDREN = :container_groups
 
+  def event_where_clause(assoc = :ems_events)
+    case assoc.to_sym
+    when :ems_events
+      # TODO: improve relationship using the id
+      ["container_namespace = ? AND container_group_name IN (?) AND #{events_table_name(assoc)}.ems_id = ?",
+       container_project.name, container_groups.pluck(:name), ems_id]
```

Comment:
@zeari @moolitayer thanks for trying to improve the query. Just to mention this will change once we'll start querying with the entity id, which will be associated with the event data.